tests/third_party/scipy/line_search_test.py

```diff
@@ -22,7 +22,7 @@ def assert_wolfe(self, s, phi, derphi, c1=1e-4, c2=0.9, err_msg=""):
     phi0 = phi(0)
     derphi0 = derphi(0)
     derphi1 = derphi(s)
-    msg = "s = %s; phi(0) = %s; phi(s) = %s; phi'(0) = %s; phi'(s) = %s; %s" % (
+    msg = "s = {}; phi(0) = {}; phi(s) = {}; phi'(0) = {}; phi'(s) = {}; {}".format(
```

Comment:
This could be improved, but should be done manually.

pyupgrade does not do very large `f-string` upgrades because they can be less readable in those cases.

I'd prefer for the conversion of this into an f-string to happen in a separate PR, but happy to add a commit here.